scripts/quests/otherAreas/His_Name_is_Valgeir.lua

```diff
@@ -0,0 +1,148 @@
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getCharVar("UnendingChaseCompDay") + 2 < VanadielDayOfTheYear() or player:getCharVar("UnendingChaseCompYear") < VanadielYear() then
+                        return quest:event(86) -- His Name is Valgeir starting event.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [86] = function(player, csid, option, npc)
+                    if option == 80 then -- Accept quest option.
+                        player:setCharVar("UnendingChaseCompDay", 0)  -- Delete previous quest (Unending Chase) variables
+                        player:setCharVar("UnendingChaseCompYear", 0) -- Delete previous quest (Unending Chase) variables
+                        player:addKeyItem(xi.ki.ARAGONEU_PIZZA) --give pizza to player
+                        player:messageSpecial(mhauraID.text.KEYITEM_OBTAINED, xi.ki.ARAGONEU_PIZZA)
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepeted.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            -- TODO: Find information about the ferry free ride. NPC involved and number of times it allows for free rides.
+            -- KNOWN: It isnt mandatory to take the ferry.
+
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(87) -- Not given the pizza yet. Yes. Really.
+                end,
+            },
+        },
+
+        [xi.zone.SELBINA] =
+        {
+            ['Valgeir'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(100) -- Deliver Pizza.
+                end,
+            },
+
+            onEventFinish =
+            {
+                [100] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.ARAGONEU_PIZZA)
+                    player:messageSpecial(selbinaID.text.KEYITEM_LOST, xi.ki.ARAGONEU_PIZZA)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+
+    },
+
+    -- Section: Finish quest.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(88) -- Finish quest.
+                end,
+            },
+
+            onEventFinish =
+            {
+                [88] = function(player, csid, option, npc)
+                    player:setCharVar("HisNameIsValgeirCompDay", VanadielDayOfTheYear()) -- Set completition day of quest.
```

Comment:
If you set these vars to be in the format of the new quest system then clear when start next quest would be better way of tracking. I'm not sure if your using the Prog[][]. Way or not